File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {

    public int maxDepthBTree(TreeNode node){

     if(node == null)
      return 0; 

     return Math.max(maxDepthBTree(node.left),maxDepthBTree(node.right))+1;   
    }


    public int maxDepth(TreeNode root) {

     if(root == null)
      return 0;   
     
     return maxDepthBTree(root);
    }
}

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:

    def maxDepthTree(self,node: TreeNode | None) -> int:
        if not node:
            return 0

        return max(self.maxDepthTree(node.left),self.maxDepthTree(node.right))+1

    def maxDepth(self, root: TreeNode | None) -> int:
        return self.maxDepthTree(root)    